### PR TITLE
Define a common interface for DirectoryAssetBundle and ZipAssetStore

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@chinmaygarde

File: runtime/asset_font_selector.cc [hunk 1]
```diff
@@ -116,14 +116,19 @@ AssetFontSelector::FlutterFontAttributes::FlutterFontAttributes(
...
+
 void AssetFontSelector::parseFontManifest() {
   std::vector<uint8_t> font_manifest_data;
-  if (!directory_asset_bundle_ ||
-      !directory_asset_bundle_->GetAsBuffer(kFontManifestAssetPath,
-                                            &font_manifest_data)) {
-    if (!asset_store_ ||
-        !asset_store_->GetAsBuffer(kFontManifestAssetPath, &font_manifest_data))
-      return;
+  if (!asset_provider()->GetAsBuffer(kFontManifestAssetPath,
```
Comment (jakobr-google) [inline runtime/asset_font_selector.cc]: I don't think this behaves exactly like before. The previous code would first look in the directory, and then in the zip file if it wasn't there (or the directory wasn't configured). The new code only looks in either the directory or the zip file, with no fallback.

Comment (jason-simmons) [reply runtime/asset_font_selector.cc]: An AssetFontSelector can be constructed from either a DirectoryAssetBundle or a ZipAssetStore.  Only one of those fields will be populated.

**Comment by szakarias** [comment]
Fuchsia is in the process of moving all their apps off FLX and then the ZipAssetStore will be removed. I would say that this is not needed.

**Comment by jason-simmons** [comment]
The intent of this PR is avoiding separate cases for DirectoryAssetBundle versus ZipAssetStore in FontCollection::RegisterFonts.

I'll close this assuming that we won't need to support serving custom fonts from a ZipAssetStore.

**Pull request closed without merging**